src/components/engagement/engagement.service.ts

```diff
@@ -15,54 +26,823 @@ import {
+          canEditProject.canEditProject as canEditProject,
+          canReadCeremony.canReadCeremony as canReadCeremony,
+          canEditCeremony.canEditCeremony as canEditCeremony,
+          canReadIntern.canReadIntern as canReadIntern,
+          canEditIntern.canEditIntern as canEditIntern,
+          canReadMentor.canReadMentor as canReadMentor,
+          canEditMentor.canEditMentor as canEditMentor,
+          canReadCountryOfOrigin.canReadCountryOfOrigin as canReadCountryOfOrigin,
+          canEditCountryOfOrigin.canEditCountryOfOrigin as canEditCountryOfOrigin,
+          canReadPosition.canReadPosition as canReadPosition,
+          canEditPosition.canEditPosition as canEditPosition,
+          canReadMethodologies.canReadMethodologies as canReadMethodologies,
+          canEditMethodologies.canEditMethodologies as canEditMethodologies,
+          canReadStatus.canReadStatus as canReadStatus,
+          canEditStatus.canEditStatus as canEditStatus,        
+          canReadCompleteDate.canReadCompleteDate as canReadCompleteDate,
+          canEditCompleteDate.canEditCompleteDate as canEditCompleteDate,
+          CanReadDisbursementCompleteDate.CanReadDisbursementCompleteDate as CanReadDisbursementCompleteDate,
+          CanEditDisbursementCompleteDate.CanEditDisbursementCompleteDate as CanEditDisbursementCompleteDate,
+          canReadCommunicationsCompleteDate.canReadCommunicationsCompleteDate as canReadCommunicationsCompleteDate,
+          canEditCommunicationsCompleteDate.canEditCommunicationsCompleteDate as canEditCommunicationsCompleteDate,
+          canReadStartDate.canReadStartDate as canReadStartDate,
+          canEditStartDate.canEditStartDate as canEditStartDate,
+          canReadEndDate.canReadEndDate as canReadEndDate,
+          canEditEndDate.canEditEndDate as canEditEndDate,
+          canReadInitialEndDate.canReadInitialEndDate as canReadInitialEndDate,
+          canReadLastSuspendedAt.canReadLastSuspendedAt as canReadLastSuspendedAt,
+          canReadLastReactivatedAt.canReadLastReactivatedAt as canReadLastReactivatedAt,
+          canReadStatusModifiedAt.canReadStatusModifiedAt as canReadStatusModifiedAt,
+          canReadModifiedAt.canReadModifiedAt as canReadModifiedAt
+      `;
+    }
+    const result = await this.db
+      .query()
+      .raw(query, {
+        token: session.token,
+        requestingUserId: session.userId,
+        owningOrgId: session.owningOrgId,
+        id,
+      })
+      .first();
+
+    if (!result) {
+      throw new NotFoundException(
+        'Could not find langage engagement or internship engagement'
+      );
+    }
+
+    const ceremony = await this.ceremonyService.readOne(
+      result.ceremonyId,
+      session
+    );
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    let language: Language = {} as Language;
+    if (result.languageId) {
+      language = await this.languageService.readOne(result.languageId, session);
+    }
```

Comment:
`language` should not be an empty object. It should be a fully flushed out `Language` or `undefined`.
```suggestion
    const language = result.languageId
      ? await this.languageService.readOne(result.languageId, session)
      : undefined;
```

Same goes for the other 3 properties below.